internal/api/README.md

```
@@ -43,14 +46,71 @@ Currently, the package implements version 2 (`v2`) of the API with all endpoints
+6.  On failure, it redirects browsers or returns a 401 error for API clients.
+
+Protected endpoints use this auth middleware. The system handles browser clients (redirecting to login) and API clients (returning 401 JSON errors) appropriately.
+
+### Authentication Service Interface (Deprecated - See `auth/service.go`)
+
+The authentication service interface provides these key operations:
+
+```go
+// Service defines the interface for API service implementations.
+type Service interface {
+	// RegisterRoutes registers all API routes with the Echo group.
```

markdownlint-cli2 (0.17.2)

Hard tabs
Column: 1

(MD010, no-hard-tabs)